src/adu-shell/src/script_tasks.cpp

```diff
@@ -59,10 +59,23 @@ ADUShellTaskResult Execute(const ADUShell_LaunchArguments& launchArgs)
             {
                 filePermissionsChanged = true;
                 stat(path, &st);
-                Log_Warn(
+                Log_Error(
                     "Failed to set '%s' file permissions (expected:%d, actual: %d)", path, mode, st.st_mode & ~S_IFMT);
             }
         }
+
+        // Ensure that the script has the correct ownership.
+        struct group* grp = ADUCPAL_getgrnam(ADUC_FILE_GROUP);
+        struct passwd* p = ADUCPAL_getpwnam(ADUC_FILE_USER);
+
+        if (p != NULL && grp != NULL)
+        {
+            // Fix the ownership.
+            if (0 != ADUCPAL_chown(path, p->pw_uid, grp->gr_gid))
+            {
+                Log_Error("Failed to set '%s' file ownership to %d:%d", path, p->pw_uid, grp->gr_gid);
```

Comment:
Should it set error value (non-zero) using taskResult.SetExitStatus and essentially "goto done" (put done: label around line 92 before return) so it does not run any more logic?

Also, for Line 76, should it be Log_Error, SetExitStatus, and also goto done?

Reply:
Fixed